### DISCUSSION TITLE
Tailwind not working in remix 1.17.1

### DISCUSSION THREAD

**Discussion by mhanifmuhsin** [remix-run/remix#6631] [Q&A] [answered]
I have problem with tailwind , 
![code](https://github.com/remix-run/remix/assets/68141840/9090cbed-7304-4cf1-901a-4bd80f09d698)

<img width="627" alt="Screenshot 2023-06-17 at 16 33 19" src="https://github.com/remix-run/remix/assets/68141840/349772ea-7bd6-42ab-a691-579e198ca9f3">

How to fix it ? thank's

this is sample code [repository](https://github.com/mhanifmuhsin/remix-1.17.1-tailwind.git)

**Comment by MichaelDeBoey** [comment]
CC/ @markdalgleish

**Comment by kiliman** [comment, marked as answer]
The issue is with the `links` export in *root.tsx*. If you don't have a CSS bundle, the ternary will always return `[]`, so it doesn't include the tailwind styles. Remix is building the CSS file just fine. You can look in */public/build/_assets*.

Change `links` from
```ts
export const links: LinksFunction = () => [
  ...(cssBundleHref
    ? [
        { rel: "stylesheet", href: cssBundleHref },
        { rel: "stylesheet", href: styles },
      ]
    : []),
];
```
to this
```ts
export const links: LinksFunction = () => [{ rel: "stylesheet", href: styles }];
```

**Reply by MichaelDeBoey** [reply]
@mhanifmuhsin As @kiliman mentioned it's because of the `cssBundleHref` check: you added tailwind into that check, instead of putting it inside the root array that's returned from `links`

**Reply by mhanifmuhsin** [reply]
thank you for helping @MichaelDeBoey and @kiliman

**Comment by Mihai-github** [comment]
Can someone give me a hand and explain why it is not working correctly with any method used?
I feel like there's not even a support team for this since every method tried today is not working 100%

**Reply by kiliman** [reply]
We try to help out as much as possible. As you can see above, we were able to resolve the OPs issue on the same day.

Please show what you're trying to do and hopefully, we can help.

**Reply by Mihai-github** [reply]
Hey there, apologies for the delayed response. Good news—I managed to figure out my issue and resolved it for the time being.

I have a question that you might have some insights on. Are there any plans for Remix.js to support .module files and .scss? I haven't come across any information on this topic, and I'm quite curious about it. It can be a bit tedious to compile SCSS code to CSS separately, and having the convenience of using `:global` would be really beneficial at times.

**Comment by Disorrder** [comment]
Still not working on clear setup. I think, it's something with postcss, it doesn't process at all
Works good on remix 2! But I need to test remix 1

<img width="779" alt="image" src="https://github.com/remix-run/remix/assets/2883539/e2a66c34-1498-416a-ab6b-2c560fcfd3ea">
<img width="779" alt="image" src="https://github.com/remix-run/remix/assets/2883539/104b545a-bbae-45a3-8f9b-0091fbac874b">
<img width="387" alt="image" src="https://github.com/remix-run/remix/assets/2883539/1f300f77-02be-4fc9-85c0-c3886df65620">

**Reply by kiliman** [reply]
Remix added automatic support for Tailwind starting in Remix v1.16. Make sure you follow the directions to enable Tailwind support https://remix.run/docs/en/1.16.0/guides/styling#tailwind-css

In particular, you need to update _remix.config.js_ to enable Tailwind.
```ts
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  tailwind: true,
  // ...
};
```
